### PR TITLE
Add optimal build options to CMakeLists

### PR DESCRIPTION
**Pull request by GhettoGirl**
Provides two options when building RPCS3
 - **USE_SYSTEM_FFMPEG** BOOL (default: OFF)
 - **USE_SYSTEM_LIBPNG** BOOL (default: OFF)

These options lets the user select between the system provided and builtin libraries of ffmpeg and libpng to overcome possible system issues.

See: https://github.com/RPCS3/rpcs3/issues/1794#issuecomment-228370509

---

This is my first time I ever did CMake so the changes may not be optimal. I tested the build with all possible combinations and all works as expected. (Compilation and linking)

```
In repository root
 cmake .
 cmake -DUSE_SYSTEM_FFMPEG=ON .
 cmake -DUSE_SYSTEM_LIBPNG=ON .
 cmake -DUSE_SYSTEM_LIBPNG=ON -DUSE_SYSTEM_FFMPEG=ON .

In rpcs3 directory
 cmake .
 cmake -DUSE_SYSTEM_FFMPEG=ON .
 cmake -DUSE_SYSTEM_LIBPNG=ON .
 cmake -DUSE_SYSTEM_LIBPNG=ON -DUSE_SYSTEM_FFMPEG=ON .
```

```ldd``` was used to verify the results.

---

~~Also if someone has a better naming scheme than **WITH_SYSTEM_blahblah** than let me know :grinning:~~

### CODE REVIEW

**Comment by rcaridade145** [comment]
This conflicts a bit with pull request #1687 . @GhettoGirl don't define libpng on target libs to a specific version as it may not be the one the user has.

**Comment by GhettoGirl** [comment]
> This conflicts a bit with pull request #1687 . @GhettoGirl don't define libpng on target libs to a specific version as it may not be the one the user has.

I'm not exactly sure what you mean here. Doesn't require RPCS3 libpng 1.6 minimum?

> This is my first time I ever did CMake  so the changes may not be optimal.

Also this the very first time I ever wrote something in CMake so I have no idea what is happening. It works on my machine and on travis.

Could you please be more specific? Thanks.

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,12 @@ set(PNG_TESTS  OFF CACHE BOOL "Build tests." FORCE)
 
 add_definitions(-DCMAKE_BUILD)
 
-add_subdirectory( 3rdparty/libpng )
+# select libpng version
+option(WITH_SYSTEM_LIBPNG "libpng" OFF)
```
Comment (paulsapps) [inline CMakeLists.txt]: Not that it matters but I think USE_SYSTEM_* is more clear?

Comment (GhettoGirl) [inline CMakeLists.txt]: You are right, thanks for the suggestion. :smile:  I will change it.

**Comment by rcaridade145** [comment]
@GhettoGirl  for instance on Ubuntu it its not - https://launchpad.net/ubuntu/+source/libpng . Please look at https://github.com/RPCS3/rpcs3/pull/1687/commits/14bb39a83bd15fd9e36ad4a35be8f9ecab61ed85 and the use of  {$PNG_LIBRARIES}.

**Comment by GhettoGirl** [comment]
@rcaridade145 Thanks for the useful infos :+1: I updated the CMake files to provide just the paths which ```find_package``` sets. Now there should not be anymore to worry about.
Could you see the last commit and tell me if everything is correct?

**Comment by GhettoGirl** [comment]
I merged 223979c0881c622e7d2244399468992f9b4574e1 into the ```cmake-tweaks``` branch to resolve a merge confict in the ```CMakeLists.txt``` file.

Btw, any news on this? Would be great if this gets merged to help out other Linux users which may have problems with the builtin libpng or ffmpeg :grinning:

**Comment by rcaridade145** [comment]
@GhettoGirl My only point would be to add the changes to cellPngDec https://github.com/RPCS3/rpcs3/pull/1687/commits/14bb39a83bd15fd9e36ad4a35be8f9ecab61ed85 so it does not break if the user has not 1.6.

**Comment by GhettoGirl** [comment]
Closing this in favor of #1841

**Pull request closed without merging**